autograder/api/config.py

```diff
@@ -241,6 +241,10 @@ def get_argument_parser(
 PARAM_TARGET_SUBMISSION_OR_RECENT = APIParam('target-submission',
         'The ID of the submission (default to the most recent submission).',
         required = False)
+        
+PARAM_TARGET_SUBMISSION = APIParam('target-submission',
```

Comment:
You should be using PARAM_TARGET_SUBMISSION_OR_RECENT.
This is an issue in your API contract that exists in your backend PR.

Reply:
Since removing a submission is a very serious and permanent request, couldn't defaulting it to the most recent submission result in an undesirable outcome if the person submitting the request had a different submission in mind?

Reply:
Removing a submission can be serious, but so is `rm`.
It's a serious command that can have serious consequences, but at some level we need to trust the user
(especially since this user must at least be a grader).

I could potentially see having the idea of a "garbage can" instead of permanently deleting a submission,
but that would mean non-trivial complications in several different places.
Making the argument for it would be hard (but not impossible).
If you want to make that argument, then you will need to finish this PR first without a garbage can (so you get familiar with submission), consider how it would work in the disk DB, and then how it would work in an actual DB.